AppCenterDataStorage/AppCenterDataStorageTests/MSDataStoreTests.m

```diff
@@ -1,33 +1,162 @@
+
+  // Then
+  XCTAssertNotNil(dic);
+  XCTAssertTrue(dic[@"Content-Type"]);
+}
+
+- (void)testDocumentUrlWithTokenResultWithStringToken {
+
+  // If
+  MSTokenResult *tokenResult = [[MSTokenResult alloc] initWithString:kMSTokenTest];
+
+  // When
+  NSString *result = [MSCosmosDb documentUrlWithTokenResult:tokenResult documentId:kMSDocumentIdTest];
+
+  // Then
+  XCTAssertNotNil(result);
+}
+
+- (void)testDocumentUrlWithTokenResultWithObjectToken {
+
+  // When
+  MSTokenResult *tokenResult = [[MSTokenResult alloc] initWithDictionary:[self prepareMutableDictionary]];
+  NSString *testResult = [MSCosmosDb documentUrlWithTokenResult:tokenResult documentId:@"documentId"];
+
+  // Then
+  XCTAssertNotNil(testResult);
+  XCTAssertTrue([testResult containsString:kMSDocumentIdTest]);
+  XCTAssertTrue([testResult containsString:kMSDbAccountTest]);
+  XCTAssertTrue([testResult containsString:kMSDbNameTest]);
+  XCTAssertTrue([testResult containsString:kMSDbCollectionNameTest]);
+}
+
+- (void)testDocumentUrlWithTokenResultWithDictionaryToken {
+
+  // If
+  MSTokenResult *tokenResult = [[MSTokenResult alloc] initWithDictionary:[self prepareMutableDictionary]];
+
+  // When
+  NSString *testResult = [MSCosmosDb documentUrlWithTokenResult:tokenResult documentId:kMSDocumentIdTest];
+
+  // Then
+  XCTAssertNotNil(testResult);
+  XCTAssertTrue([testResult containsString:kMSDocumentIdTest]);
+  XCTAssertTrue([testResult containsString:kMSDbAccountTest]);
+  XCTAssertTrue([testResult containsString:kMSDbNameTest]);
+  XCTAssertTrue([testResult containsString:kMSDbCollectionNameTest]);
+}
+
+- (void)testPerformCosmosDbAsyncOperationWithHttpClientWithAdditionalParams {
+
+  // If
+  MSCosmosDbIngestion *httpClient = [[MSCosmosDbIngestion alloc] init];
+  MSTokenResult *tokenResult = [[MSTokenResult alloc] initWithDictionary:[self prepareMutableDictionary]];
+  NSMutableDictionary *additionalHeaders = [NSMutableDictionary new];
+  additionalHeaders[@"Foo"] = @"Bar";
+  __block BOOL completionHandlerCalled = NO;
+  XCTestExpectation *completeExpectation = [self expectationWithDescription:@"Task finished"];
+  MSCosmosDbCompletionHandler handler = ^(NSData *_Nullable data, NSError *_Nullable error) {
+    completionHandlerCalled = YES;
+    [completeExpectation fulfill];
+  };
+
+  // When
+  [MSCosmosDb performCosmosDbAsyncOperationWithHttpClient:httpClient
+                                              tokenResult:tokenResult
+                                               documentId:kMSDocumentIdTest
+                                               httpMethod:kMSHttpMethodGet
+                                                     body:nil
+                                        additionalHeaders:additionalHeaders
+                                        completionHandler:handler];
+  [self waitForExpectationsWithTimeout:5 handler:nil];
+
+  // Then
```

Comment:
I wonder if we also need to check if `httpClient` is executed with the correct arguments that we're expecting, not only to check if it was called.